src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -557,17 +558,25 @@ private void copyKeyAndTitle() {
                 return;
             }
 
-            Globals.clipboardManager.setContent(sb.toString());
+            final String copiedKeysAndTitles = sb.toString();
+            Globals.clipboardManager.setContent(copiedKeysAndTitles);
 
             if (copied == bes.size()) {
                 // All entries had keys.
-                output((bes.size() > 1 ? Localization.lang("Copied keys") : Localization.lang("Copied key")) + '.');
+                output(Localization.lang("Copied") + " '" + shortenDialogMessage(copiedKeysAndTitles) + "'.");
             } else {
                 output(Localization.lang("Warning: %0 out of %1 entries have undefined BibTeX key.", Integer.toString(bes.size() - copied), Integer.toString(bes.size())));
             }
         }
     }
 
+    private String shortenDialogMessage(String dialogMessage) {
```

Comment:
Maybe move `shortenDialogMessage` to a helper class (add `OverrunStyle` as an additional paramater) and reuse it below in CopyBibTeXKeyAndLinkAction ?

Reply:
revisiting the code, I decided to move the message to the JabRefDialogService (formerly FXDialogService)